src/ray/gcs/redis_context.cc

```diff
@@ -324,14 +325,15 @@ Status AuthenticateRedis(redisAsyncContext *context,
 
 void RedisAsyncContextDisconnectCallback(const redisAsyncContext *context, int status) {
   RAY_LOG(DEBUG) << "Redis async context disconnected. Status: " << status;
-  // Reset raw 'redisAsyncContext' to nullptr because hiredis will release this context.
-  reinterpret_cast<RedisAsyncContext *>(context->data)->ResetRawRedisAsyncContext();
+  // Reset 'RedisAsyncContext' to nullptr because hiredis will release the raw async
+  // context.
+  reinterpret_cast<RedisContext *>(context->data)->ResetAsyncContext();
```

Comment:
Do we need to call `release()` of the underlying raw async context as before?

Reply:
`reset()` will internally call `release()`?

Reply:
1. `reset` will delete the underlying object, while `release` will not.  
2. `release` returns the underlying object, which is not needed in this case.  
3. Therefore, I think `reset` would be better here.  

- https://stackoverflow.com/questions/25609457/does-unique-ptrrelease-call-the-destructor  
- https://en.cppreference.com/w/cpp/memory/unique_ptr/release  
- https://en.cppreference.com/w/cpp/memory/unique_ptr/reset